sklearn/linear_model/tests/test_ridge.py

```diff
@@ -836,6 +837,65 @@ def test_ridge_fit_intercept_sparse():
         assert_array_almost_equal(dense.coef_, sparse.coef_)
 
 
+@pytest.mark.parametrize('return_intercept', [False, True])
+@pytest.mark.parametrize('sample_weight', [None, np.ones(1000)])
+@pytest.mark.parametrize('arr_type', [np.array, sp.csr_matrix])
+@pytest.mark.parametrize('solver', ['auto', 'sparse_cg', 'cholesky', 'lsqr',
+                                    'sag', 'saga'])
+def test_ridge_regression_check_arguments_validity(return_intercept,
+                                                   sample_weight, arr_type,
+                                                   solver):
+    """check if all combinations of arguments give valid estimations"""
+
+    # test excludes 'svd' solver because it raises exception for sparse inputs
+
+    X = np.random.rand(1000, 3)
+    true_coefs = [1, 2, 0.1]
+    y = np.dot(X, true_coefs)
+    X_testing = arr_type(X)
+
+    target = ridge_regression(X_testing, y, 1,
+                              solver=solver,
+                              sample_weight=sample_weight,
+                              return_intercept=return_intercept
+                              )
+    if return_intercept:
+        coef, intercept = target
+        assert_array_almost_equal(coef, true_coefs, decimal=1)
+        assert_array_almost_equal(intercept, 0, decimal=1)
+    else:
+        assert_array_almost_equal(target, true_coefs, decimal=1)
+
+
+def test_ridge_regression_warns_with_return_intercept():
+    # return_itercept is only supported by sag/saga
+
+    X, y = make_regression(n_samples=1000, n_features=2, n_informative=2,
+                           bias=10., random_state=42)
+
+    for solver in ['sparse_cg', 'cholesky', 'svd', 'lsqr', 'saga']:
+        with pytest.warns(UserWarning) as record:
+            target = ridge_regression(X, y, 1,
+                                      solver=solver,
+                                      return_intercept=True
+                                      )
+            assert len(target) == 2
+            assert target[0].shape == (2,)
+
+        for r in record:
+            r.message.args[0].startswith("return_intercept=True is only")
+
```

Comment:
you forgot an `assert` here. But you'd not need these lines if you do as above.

Reply:
I removed these lines